src/wallet/wallet.ts

```diff
@@ -464,45 +461,51 @@ class HathorWalletServiceWallet extends EventEmitter implements IHathorWallet {
-    let xpub: string;
-    let authXpub: string;
+  async generateCreateWalletAuthData(
+    accessData: IWalletAccessData,
+    pinCode: string
+  ): Promise<CreateWalletAuthData> {
+    let privKey: bitcore.HDPrivateKey;
     let privKeyAccountPath: bitcore.HDPrivateKey;
     let authDerivedPrivKey: bitcore.HDPrivateKey;
 
     const now = Date.now();
     const timestampNow = Math.floor(now / 1000); // in seconds
 
-    if (this.seed) {
+    if (accessData.acctPathKey) {
+      const acctKey = decryptData(accessData.acctPathKey, pinCode);
+      privKeyAccountPath = bitcore.HDPrivateKey(acctKey);
+    } else if (this.seed) {
+      // Derive from seed present in memory
       // getXPrivKeyFromSeed returns a HDPrivateKey on the root path
-      const privKey: bitcore.HDPrivateKey = walletUtils.getXPrivKeyFromSeed(this.seed, {
-        passphrase: this.passphrase,
-        networkName: this.network.name,
-      });
-      // getXPubKeyFromSeed returns a xpubkey on the account level path
-      xpub = walletUtils.getXPubKeyFromSeed(this.seed, {
+      privKey = walletUtils.getXPrivKeyFromSeed(this.seed, {
         passphrase: this.passphrase,
         networkName: this.network.name,
       });
-      authXpub = HathorWalletServiceWallet.getAuthXPubKeyFromSeed(this.seed, {
-        passphrase: this.passphrase,
-        networkName: this.network.name,
-      });
-      privKeyAccountPath = walletUtils.deriveXpriv(privKey, "0'");
-      authDerivedPrivKey = HathorWalletServiceWallet.deriveAuthPrivateKey(privKey);
+      privKeyAccountPath = privKey.deriveNonCompliantChild(P2PKH_ACCT_PATH);
     } else if (this.xpriv) {
-      // this.xpriv is already on the account derivation path
       privKeyAccountPath = bitcore.HDPrivateKey(this.xpriv);
-      xpub = privKeyAccountPath.xpubkey;
+    } else {
+      throw new Error('Cannot derive account path key');
+    }
+    const xpub = privKeyAccountPath.xpubkey;
 
-      // If the wallet is being loaded from the xpriv, we assume we already have the authXPriv on storage, so just fetch it
-      authDerivedPrivKey = bitcore.HDPrivateKey.fromString(
-        await this.storage.getAuthPrivKey(pinCode)
-      );
-      authXpub = authDerivedPrivKey.xpubkey;
+    if (accessData.authKey) {
+      const authKey = decryptData(accessData.authKey, pinCode);
+      authDerivedPrivKey = bitcore.HDPrivateKey(authKey);
+    } else if (this.seed) {
+      // Derive from seed present in memory
+      if (!privKey) {
+        privKey = walletUtils.getXPrivKeyFromSeed(this.seed, {
+          passphrase: this.passphrase,
+          networkName: this.network.name,
+        });
+      }
+      authDerivedPrivKey = privKey.deriveNonCompliantChild(WALLET_SERVICE_AUTH_DERIVATION_PATH);
     } else {
-      throw new Error('generateCreateWalletAuthData called without seed or xpriv in memory.');
+      throw new Error('Cannot derive auth path key');
```

Comment:
Shouldn't we have an `else if (this.xpriv)` as before? To fetch the `authDerivedPrivKey` from storage.

```
 // If the wallet is being loaded from the xpriv, we assume we already have the authXPriv on storage, so just fetch it
      authDerivedPrivKey = bitcore.HDPrivateKey.fromString(
        await this.storage.getAuthPrivKey(pinCode)
      );
```